modules/hal_nxp/CMakeLists.txt

```diff
@@ -4,7 +4,9 @@
 # SPDX-License-Identifier: Apache-2.0
 #
 
-add_subdirectory(${ZEPHYR_CURRENT_MODULE_DIR} hal_nxp)
-add_subdirectory_ifdef(CONFIG_USB_DEVICE_DRIVER usb)
+if(CONFIG_USE_HAL_NXP)
```

Comment:
can we use `if(CONFIG_HAS_MCUX OR CONFIG_HAS_IMX_HAL OR CONFIG_HAS_S32_HAL)` instead of defining a new Kconfig option? This is already being done here https://github.com/zephyrproject-rtos/hal_nxp/blob/master/CMakeLists.txt#L1